src/redisraft.c

```diff
@@ -591,76 +591,80 @@ static bool handleInterceptedCommands(RedisRaftCtx *rr,
-     * are enabled schedule the request to be processed when we have a guarantee
-     * we're still a leader. Otherwise, just process the reads.
-     *
-     * Normally we can expect a single command in the request, unless it is a
-     * MULTI/EXEC transaction in which case all queued commands are handled at
-     * once.
-     */
-    unsigned int cmd_flags = CommandSpecGetAggregateFlags(cmds, CMD_SPEC_WRITE);
-    if (cmd_flags & CMD_SPEC_UNSUPPORTED) {
-        RedisModule_ReplyWithError(ctx, "ERR not supported by RedisRaft");
-        return RR_ERROR;
-    } else if (cmd_flags & CMD_SPEC_READONLY && !(cmd_flags & CMD_SPEC_WRITE)) {
-        if (rr->config->quorum_reads) {
-            RaftReq *req = RaftReqInit(ctx, RR_REDISCOMMAND);
-            RaftRedisCommandArrayMove(&req->r.redis.cmds, cmds);
-            raft_queue_read_request(rr->raft, handleReadOnlyCommand, req);
-        } else {
-            /* Wait until the new leader applies an entry from the current term.
-             * Otherwise, we might process a request before replaying logs.
-             * The state machine will be in an older state. Reading from it
-             * might look like going backward in time. */
-            raft_term_t term = raft_get_current_term(rr->raft);
-            if (term != rr->snapshot_info.last_applied_term) {
-                replyClusterDown(ctx);
-                return RR_ERROR;
-            }
-
+    if (rr->config->quorum_reads) {
+        RaftReq *req = RaftReqInit(ctx, RR_REDISCOMMAND);
+        RaftRedisCommandArrayMove(&req->r.redis.cmds, cmds);
+        raft_queue_read_request(rr->raft, handleReadOnlyCommand, req);
+    } else {
+        /* Wait until the new leader applies an entry from the current term.
+         * Otherwise, we might process a request before replaying logs.
+         * The state machine will be in an older state. Reading from it
+         * might look like going backward in time. */
+        raft_term_t term = raft_get_current_term(rr->raft);
+        if (term == rr->snapshot_info.last_applied_term) {
             RaftExecuteCommandArray(rr, ctx, ctx, cmds);
+        } else {
+            replyClusterDown(ctx);
         }
-        return RR_OK;
     }
+}
 
+static void handleRedisCommandAppendLog(RedisRaftCtx *rr, RedisModuleCtx *ctx, RaftRedisCommandArray *cmds) {
     RaftReq *req = RaftReqInit(ctx, RR_REDISCOMMAND);
     RaftRedisCommandArrayMove(&req->r.redis.cmds, cmds);
 
     raft_entry_t *entry = RaftRedisCommandArraySerialize(&req->r.redis.cmds);
     entry->id = rand();
     entry->type = RAFT_LOGTYPE_NORMAL;
     entryAttachRaftReq(rr, entry, req);
+
     int e = raft_recv_entry(rr->raft, entry, &req->r.redis.response);
     if (e != 0) {
         replyRaftError(ctx, e);
         entryDetachRaftReq(rr, entry);
         raft_entry_release(entry);
         RaftReqFree(req);
-        return RR_ERROR;
+    } else {
+        raft_entry_release(entry);
```

Comment:
As both paths are calling `raft_entry_release(entry)`, we can call this unconditionally but I think I find more readable when we separate error paths from the success path and keep success path in the main flow. (Not in an "if check" if possible).  Previously, when we get an error, we were calling bunch of functions and we were returning from the error case. I find it easier to understand.

Reply:
ok, reverted else and put void return back in